analytics-recordings.php

```diff
@@ -0,0 +1,197 @@
+$table->is_downloading($download);
+
+if (!$download) {
+    // Only print headers if not asked to download data.
+    $PAGE->set_title($title);
+    $PAGE->set_cacheable(false);
+    $PAGE->set_heading($heading);
+    $PAGE->navbar->add(get_string('view_section_title_analytics', 'bigbluebuttonbn'), $pageurl);
+    echo $OUTPUT->header();
+}
+
+// Analytics Report Query.
+$wheresql = "meta IS NOT NULL
+             AND log = :log
+             AND userid IS NOT NULL";
+$params = ['log' => BIGBLUEBUTTONBN_LOG_EVENT_CREATE];
+if (!empty($bn) && !empty($bigbluebuttonbn->id)) {
+    $wheresql .= " AND bigbluebuttonbnid = :bigbluebuttonbnid";
+    $params['bigbluebuttonbnid'] = $bigbluebuttonbn->id;
+}
+$table->set_sql('id, meta, userid', "{bigbluebuttonbn_logs}", $wheresql, $params);
+$table->define_baseurl($pageurl);
+
+// Add cloned table modifying the query for past week instead.
+$pastweektable = clone $table;
+$wheresqlwithweekconstraint = $wheresql;
+$wheresql .= " AND timecreated > :timecreated";
+$params['timecreated'] = time() - WEEKSECS;
+$pastweektable->set_sql('id, meta, userid', "{bigbluebuttonbn_logs}", $wheresql, $params);
+$pastweektable->setup();
+$pastweektable->query_db(0); // No limit, fetch all rows.
+$pastweektable->pageable(false);
+$pastweektable->close_recordset();
+$pastweekmeetingcreateevents = $pastweektable->rawdata;
+
+$output = "";
+
+// Prepare attendee data for Attention Box.
+$table->setup();
+$table->query_db(0); // No limit, fetch all rows.
+$table->pageable(false);
+$table->close_recordset();
+$meetingcreateevents = $table->rawdata;
+
+if (isset($meetingsummary->meta)) {
+    $meetingsummary->meta = json_decode($meetingsummary->meta);
+}
+
+if (!$download && !empty($meetingcreateevents)) {
+    $tables = "";
+    $output .= $OUTPUT->heading(get_string('view_analytics_heading_recordings', 'bigbluebuttonbn'), 3);
+
+    // Function to help render the table given the data ($events) and a table name.
+    $createoverviewtable = function($tablename, $events) {
+        // Helper function to return a reducer that will sum an array of objects
+        // given a key. This will default to zero for the value if the value does
+        // not exist in the object.
+        $sumkeyfunc = function($key) use($events) {
+            return array_reduce($events, function($acc, $row) use ($key) {
+                $data = json_decode($row->meta);
+                return $acc + ($data->{$key} ?? 0);
+            }, 0);
+        };
+
+        $processingdurationtotalinseconds = $sumkeyfunc('processingduration') / 1000;
+        $filesizetotal = $sumkeyfunc('filesize');
+        $overviewtable = new html_table();
+        $overviewtable->caption = $tablename;
+        $overviewtable->attributes['class'] = 'w-auto admintable generaltable mr-4 table-bordered';
+        $htmlparams = ['class' => 'text-center'];
+        $data = [
+            [get_string('view_analytics_total_recordings', 'bigbluebuttonbn'), count($events)],
```

Comment:
You can do a complete get_string call and get an array back, then access by index. might read a little nicer